From cf2fb42976fa0b30aa58b096756bfe74ba3c4092 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 17 Oct 2022 16:30:32 +0200
Subject: [PATCH 05/10] FROM ARCHLINUX: support building from dkms
X-Developer-Signature: v=1; a=openpgp-sha256; l=1456; i=xtexchooser@duck.com;
 h=from:subject; bh=xaE43/NlZnLKMRTive/VKZViovIuaB1UAW77tluHxjg=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDGkvVLY28G8/Pf38Po75v2ez3Kxw2NU/jXPLVoXpp2OvB
 yseDXDS7yhlYRDjYpAVU2QpMmzwZtVJ5xddVi4LM4eVCWQIAxenAEykRpbhf6xTqF1Roey9gme2
 u1g03vlcL9H8lyQz13/Ftu/WgjKsCYwMy1XafD22pL/+2Mi/0qnlaNen0OV5rBlvbizS3RaxLj2
 AGwA=
X-Developer-Key: i=xtexchooser@duck.com; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

---
 src/VBox/HostDrivers/linux/Makefile | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/VBox/HostDrivers/linux/Makefile b/src/VBox/HostDrivers/linux/Makefile
index c32cb136b23f..24ce3d0e3321 100644
--- a/src/VBox/HostDrivers/linux/Makefile
+++ b/src/VBox/HostDrivers/linux/Makefile
@@ -33,25 +33,25 @@
 # SPDX-License-Identifier: GPL-3.0-only OR CDDL-1.0
 #
 
-ifneq ($(KERNELRELEASE),)
+ifneq ($(KBUILD_EXTMOD),)
 
-# Building from kBuild (make -C <kernel_directory> M=`pwd`),
-# or inside a kernel source tree.
+# Building from kBuild (make -C <kernel_directory> M=`pwd`).
+# KBUILD_EXTMOD is set to $(M) in this case.
 
 obj-m = vboxdrv/
- ifneq ($(wildcard $(CURDIR)/vboxnetflt/Makefile),)
+ ifneq ($(wildcard $(KBUILD_EXTMOD)/vboxnetflt/Makefile),)
 obj-m += vboxnetflt/
  endif
- ifneq ($(wildcard $(CURDIR)/vboxnetadp/Makefile),)
+ ifneq ($(wildcard $(KBUILD_EXTMOD)/vboxnetadp/Makefile),)
 obj-m += vboxnetadp/
  endif
- ifneq ($(wildcard $(CURDIR)/vboxpci/Makefile),)
+ ifneq ($(wildcard $(KBUILD_EXTMOD)/vboxpci/Makefile),)
 obj-m += vboxpci/
  endif
 
-else # ! KERNELRELEASE
+else # ! KBUILD_EXTMOD
 
-# convenience Makefile without KERNELRELEASE
+# convenience Makefile without KBUILD_EXTMOD
 
 ifndef SUDO
  ifneq ($(shell id -u),0)
@@ -244,5 +244,5 @@ load: unload
 		fi; \
 	done
 
-endif # ! KERNELRELEASE
+endif # ! KBUILD_EXTMOD
 
-- 
2.46.0

